.github/CONTRIBUTING.md

```diff
@@ -98,6 +105,108 @@ Focused component needs to be clearly visible. This is handled in Stardust by fo
 
 In addition to basic focus handling, specific keyboard handlers can be added to the behaviors. These keyboard handlers call actions defined in the components, when corresponding keys are pressed by the user. For reference, see `keyActions` in [MenuItemBehavior][12] and `actionHandlers` in [MenuItem component][13].
 
+## Packages
+
+We are using [Lerna][14] to manage our packages and [Yarn Workspaces][15] to link them.
+
+### Add a new package
+
+#### Run `lerna create`
+ 
+You should to run `lerna create` command to create a new package
+
+- we are using `@stardust-ui` namespace on NPM to publish our packages
+- the directory name should not contain any namespace prefix and can be prefixed with the library name if the
+  implementation is not framework agnostic
+- please provide a meaningful description to a package in the matched field
+- use `https://github.com/stardust-ui/react/tree/master/packages/__DIRECTORY_NAME__` as `homepage`
+- our packages are published with MIT license, please follow it until you will have a specific requirements
+
+```sh
+lerna create @stardust-ui/react-proptypes react-proptypes
+```
+
+##### Example input
+```
+lerna notice cli v3.11.1
+package name: (@stardust-ui/react-proptypes)
+version: (0.21.1)
+description: Set of custom reusable propTypes for React components.
+keywords:
+homepage: https://github.com/stardust-ui/react/tree/master/packages/react-proptypes
+license: (ISC) MIT
+entry point: (lib/react-proptypes.js)
+git repository: (https://github.com/stardust-ui/react.git)
+```
+
+#### Update `package.json`
+
+After a package will be created we need to add necessary changes to a newly created `package.json`.
+These changes are required to get working package publishing and internal tooling.
+
+```diff
+-  "directories": {
+-    "lib": "lib",
+-    "test": "__tests__"
+-  },
+-  "files": [
+-    "lib"
+-  ],
++  "jsnext:main": "dist/es/index.js",
++  "main": "dist/commonjs/index.js",
++  "module": "dist/es/index.js",
++  "types": "dist/es/index.d.ts",
++  "sideEffects": false,
++  "files": [
++    "dist"
++  ],
+```
+
+```diff
+-  "scripts": {
+-    "test": "echo \"Error: run tests from root\" && exit 1"
+-  },
++  "scripts": {
++    "build": "gulp bundle:package:no-umd --package react-proptypes"
++  },
+```
+
+Don't forget to fill a correct directory name, you can also use `gulp bundle:package` to bundle your package with UMD.
+
+#### Create `tsconfig.json`
+
+If your package uses TypeScript, please also create a new `tsconfig.json` and place it in `packages/__DIRECTORY_NAME__`.
+An example config:
+
+```json
+{
+  "extends": "../../build/tsconfig.common",
```

Comment:
lets make it explicit in the text that this common configuration file is meant to be extended

Reply:
Added:
`This config will extend a common TS config that is used in all packages. You can add specific options for the package here.`